### PR TITLE
Add possibility to decide reporting for metrics

### LINKED ISSUE #390: Add possibility to turn off writing metrics to file but keep producing prometheus metrics
Currently metrics are either disabled or enabled but there are some use cases that would like to only do one or the other.

### PR DESCRIPTION
**Pull request by masokol**
Closes #390

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 ## Version 4.0.0
 
-* Disable driver metrics if statistics.enabled=false - #391
+* Add possibility to decide reporting for metrics - Issue #390
+* Disable driver metrics if statistics.enabled=false - Issue #391
```
Comment (itskarlsson) [inline CHANGES.md:6]: Should be first

Comment (masokol) [reply CHANGES.md:6]: This is 390, the 391 is already merged.

File: application/src/main/resources/ecc.yml [hunk 2]
```diff
@@ -167,6 +167,16 @@ repair:
 
 statistics:
   enabled: true
+  ##
+  ## How metrics should be reported
```
Comment (itskarlsson) [inline application/src/main/resources/ecc.yml:171]: Some comment about what is the behavior if all are disabled but statistics is true would be good here. As well as if statistics is false and all are enabled.

File: application/src/test/resources/all_reporting_disabled.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#
+# Copyright 2020 Telefonaktiebolaget LM Ericsson
```
Comment (itskarlsson) [inline application/src/test/resources/all_reporting_disabled.yml:2]: 2022

**Review by itskarlsson — APPROVED** [review]

**Pull request merged**